src/plone/app/multilingual/browser/translate.py

```diff
@@ -57,7 +58,19 @@ def __call__(self):
         ):
             return _("Need a field")
         else:
-            manager = ITranslationManager(self.context)
+            context_uid = self.request.form.get("context_uid", None)
+            if context_uid is None:
+                # try with context if no translation uid is present
+                manager = ITranslationManager(self.context)
+            else:
+                catalog = getToolByName(self.context, "portal_catalog")
+                brains = catalog(UID=context_uid)
+                if len(brains):
+                    context = brains[0].getObject()
```

Comment:
This is what uuidToObject is for (This is what https://github.com/plone/plone.app.uuid/blob/master/plone/app/uuid/utils.py#L76)